projects/plugins/social/src/class-jetpack-social.php

```diff
@@ -167,11 +182,27 @@ public function initial_state() {
 				'connections' => $publicize->get_all_connections_for_user(), // TODO: Sanitize the array
 				'adminUrl'    => esc_url_raw( $publicize->publicize_connections_url( 'jetpack-social-connections-admin-page' ) ),
 			),
-			'sharesData'      => ! is_wp_error( $shares ) ? $shares : null,
-			'showNudge'       => $show_nudge,
+			'sharesData'      => $this->get_shares_info(),
+			'showNudge'       => ! $this->has_paid_plan(),
 		);
 	}
 
+	/**
+	 * Returns a boolean as to whether we have a plan that supports
+	 * sharing beyond the free limit.
+	 *
+	 * It also caches the result to make sure that we don't call the API
+	 * more than once a request.
+	 *
+	 * @returns boolean True if the site has a plan that supports a higher share limit.
+	 */
+	public function has_paid_plan() {
+		static $refresh_plan_from_wpcom = true;
+		$has_plan                       = Current_Plan::supports( 'social-shares-1000', $refresh_plan_from_wpcom );
+		$refresh_plan_from_wpcom        = false;
+		return $has_plan;
```

Comment:
Would it make more sense to just cache the `$has_plan` value and return that early instead?

Reply:
Yeah probably!

Reply:
Addressed in 7428ce068b